src/test/run-make/track-path-dep-info/macro_def.rs

```diff
@@ -1,11 +1,21 @@
-#![feature(track_path)]
+#![feature(proc_macro_tracked_env,proc_macro_tracked_path)]
 #![crate_type = "proc-macro"]
 
 extern crate proc_macro;
 use proc_macro::*;
 
+use std::str;
+
 #[proc_macro]
 pub fn access_tracked_paths(_: TokenStream) -> TokenStream {
     tracked_path::path("emojis.txt");
+
+    // currently only valid utf-8 paths are supported
+    let invalid = [1_u8, 2,123, 254, 0, 0, 1, 1];
+    let invalid: &str = unsafe {
+        str::from_utf8_unchecked(&invalid[..])
```

Comment:
This is undefined behavior though.
I don't think UB is something that we should be testing.

Reply:
In https://github.com/rust-lang/rust/pull/87173 it would be possible to safely construct a `PathBuf` that is not a valid UTF-8 path and then pass it to libproc_macro.